src/server/controllers/AuthController.js

```diff
@@ -0,0 +1,84 @@
+import {
+  generateToken,
+  validateSignupPayload
+} from '../utils/auth';
+import { extractUserInfo, getUser } from '../utils/user';
+import models from '../models';
+
+/**
+ * @class AuthController
+ */
+class AuthController {
+  /**
+   * Create a user
+   * @method
+   * @memberof AuthController
+   * @static
+   * @return {function} Express middleware function that creates
+   * new user and sends response to client
+   */
+  static createUser() {
+    return (req, res, next) => {
+      try {
+        validateSignupPayload(req.body);
+      } catch (err) {
+        err.code = 422;
+        return next(err);
+      }
+      const { username, email, password } = req.body;
+      return models.User.create({ username, email, password })
+        .then((user) => {
+          req.user = extractUserInfo(user);
+          return next();
+        })
+        .catch(err => next(err));
+    };
+  }
+
+  /**
+   * Authenticate a user with username and password
+   * @method
+   * @memberof UserController
+   * @static
+   * @return {function} Express middleware function that
+   * validates username and password  and sends token to client
+   */
+  static authenticateUser() {
+    return (req, res, next) => {
+      const { username, password } = req.body;
+      return getUser({ username })
+        .then(user =>
+          user.verifyPassword(password)
+            .then((passwordIsValid) => {
+              if (passwordIsValid) {
+                const rsaKey = process.env.PRIVATE_KEY;
+                return generateToken(user, rsaKey);
+              }
+              const error = new Error('Invalid Password');
+              error.code = 422;
```

Comment:
Am curious to why you decided to return a `422` when you have an invalid password

Reply:
You  used `err` before but now you are using `error`, you should try and be consistent when defining variables